connectors/stash/src/main/resources/static/discovery/metadata.hal

```diff
@@ -0,0 +1,9 @@
+{
+  "authorization_header": "x-stash-authorization",
+  "backend_base_url_header": "x-stash-base-url",
+  "fields": {
+    "pr_email_subject": {
+      "regex": "(([a-zA-Z0-9]+)\/([a-zA-Z0-9-]+) - Pull request #([0-9]+): [a-zA-Z0-9- .\\()#$&@+=_|!]*)"
```

Comment:
We can reduce this.  We only care about the first 3 pieces, so I think you can get away with:

```
"regex": "(([a-zA-Z0-9]+)\/([a-zA-Z0-9-]+) - Pull request #([0-9]+):[ ])"
```

Reply:
The client is not coping with multiple capture groups the way you want (they are sending whatever the last capture group is, so in this case, it would only send in the PR#).

You'll have to flag the inner ones as non-capturing (which will make your regex in the json different than the regex in the java code -- this also makes the regex unit test have to do what I mention in the other comment: expectList has to match what the client sends in).

Non-capturing group is specified with `?:`

```javascript
{
    "regex": "((?:[a-zA-Z0-9]+)/(?:[a-zA-Z0-9-]+) - Pull request #(?:[0-9]+):[ ])"
}
```

Reply:
Changed the regex to ```(([a-zA-Z0-9]+)\/([a-zA-Z0-9-]+) - Pull request #([0-9]+):[ ])```..